web_widget_text_markdown/static/src/js/web_widget_text_markdown.js

```diff
@@ -0,0 +1,181 @@
+                extensions: [...clss_bindings, "table", "footnotes", "toc"],
+                emoji: true,
+                underline: true,
+                tablesHeaderId: true,
+                omitExtraWLInCodeBlocks: true,
+                noHeaderId: true,
+                prefixHeaderId: true,
+                rawPrefixHeaderId: true,
+                ghCompatibleHeaderId: true,
+                rawHeaderId: true,
+                headerLevelStart: false,
+                parseImgDimensions: true,
+                simplifiedAutoLink: true,
+                literalMidWordUnderscores: false,
+                literalMidWordAsterisks: true,
+                strikethrough: true,
+                tables: true,
+                ghCodeBlocks: true,
+                tasklists: true,
+                smoothLivePreview: true,
+                smartIndentationFix: true,
+                disableForced4SpacesIndentedSublists: true,
+                simpleLineBreaks: true,
+                requireSpaceBeforeHeadingText: true,
+                ghMentions: true,
+                ghMentionsLink: "https://github.com/{u}",
+                encodeEmails: true,
+                openLinksInNewWindow: true,
+                backslashEscapesHTMLTags: true,
+                completeHTMLDocument: true,
+                metadata: true,
+                splitAdjacentBlockquotes: true,
+            });
+        },
+        _renderEdit: function() {
+            var self = this;
+            return $.when(this._super()).then(function() {
+                if (
+                    self.mode === "edit" &&
+                    self.$type === "html" &&
+                    self.$widget &&
+                    self.$widget === "bootstrap_markdown" &&
+                    self.value
+                ) {
+                    self.value = self._getMarkdownValue(self.value);
+                }
+            });
+        },
+
+        _prepareInput: function() {
+            var $input = this._super.apply(this, arguments);
+
+            _.defer(
+                function($elm) {
+                    $input.removeClass(this.className);
+                    $input.wrap(
+                        _.str.sprintf("<div class='%s'></div>", this.className)
+                    );
+                    $elm.markdown(this._getMarkdownOptions());
+                    this.$markdown = $elm.data("markdown");
+                    this.$markdown.setContent(this.value || "");
+                }.bind(this),
+                $input
+            );
+            return $input;
+        },
+        _getMarkdownValue: function(value) {
+            return this.shw_render_html.makeMarkdown(this._formatValue(value));
+        },
+        _getHtmlValue: function(value) {
+            return this.shw_render_html.makeHtml(this._formatValue(value));
+        },
+
+        _renderReadonly: function() {
+            this.$el.html(this._getHtmlValue(this.value));
+        },
```

Comment:
For 'Implement Coupa' issue among others, it's sometimes losing italic/bold markup, and sometimes not.

Also, it's not always display existing italics in the text.